snapcraft/parts/plugins/matter_sdk_plugin.py

```diff
@@ -111,47 +111,48 @@ def get_build_snaps(self) -> Set[str]:
 
         if self.snap_arch == "arm64":
             commands.extend(
                 [
-                    f"wget --no-verbose https://github.com/project-chip/zap/releases/download/"
-                    f"{options.matter_zap_version}/zap-linux-{self.snap_arch}.zip",
+                    "wget --no-verbose https://github.com/project-chip/zap/releases/download/"
+                    f"{options.matter_sdk_zap_version}/zap-linux-{self.snap_arch}.zip",
                     f"unzip -o zap-linux-{self.snap_arch}.zip",
                     "echo 'export ZAP_INSTALL_PATH=$PWD'",
                 ]
             )
 
-        # Clone Matter repository if not present
+        # Clone Matter SDK repository if not present
         commands.extend(
             [
-                "if [ ! -d matter ]; then",
-                f"    git clone --depth 1 -b {options.matter_sdk_version} {MATTER_REPO} matter;",
-                "else",
-                "    echo 'Matter repository already exists, skip clone';",
-                "fi",
-                "cd matter;",
-            ]
+            "if [ ! -d matter ]; then",
+            "    git init matter-sdk",
+            "    cd matter-sdk",
+            f"   git remote add origin {MATTER_SDK_REPO}",
+            f"   git fetch --depth 1 origin {options.matter_sdk_version}",
+            "    git checkout FETCH_HEAD",
+            "else",
+            "    echo 'Matter SDK repository already exists, skip clone'",
+            "    cd matter-sdk;",
+            "fi",]
         )
 
         # Checkout submodules for Linux platform
         commands.extend(["scripts/checkout_submodules.py --shallow --platform linux"])
 
-        """
-        The project writes its data to /tmp which isn't persisted.
+        # The project writes its data to /tmp which isn't persisted.
 
-        Setting TMPDIR env var when running the app isn't sufficient as
-         chip_[config,counter,factory,kvs].ini still get written under /tmp.
-         The chip-tool currently has no way of overriding the default paths to
-         storage and security config files.
+        # Setting TMPDIR env var when running the app isn't sufficient as
+        # chip_[config,counter,factory,kvs].ini still get written under /tmp.
+        # The chip-tool currently has no way of overriding the default paths to
+        # storage and security config files.
 
-        Snap does not allow bind mounting a persistent directory on /tmp,
-         so we need to replace it in the source with another path, e.g. /mnt.
-         See the top-level layout definition which bind mounts a persisted
-         directory within the confined snap space on /mnt.
-        """
+        # Snap does not allow bind mounting a persistent directory on /tmp,
+        # so we need to replace it in the source with another path, e.g. /mnt.
+        # See the top-level layout definition which bind mounts a persisted
+        # directory within the confined snap space on /mnt.
```

Comment:
These comments don't make much sense, they need to be modified to fit this context.